openpype/hosts/max/api/lib.py

```diff
@@ -322,18 +323,87 @@ def is_headless():
 
 
 @contextlib.contextmanager
-def viewport_camera(camera):
+def viewport_setup_updated(camera):
+    """Function to set viewport camera during context
+    ***For 3dsMax 2024+
+    Args:
+        camera (str): viewport camera for review render
+    """
     original = rt.viewport.getCamera()
+    has_autoplay = rt.preferences.playPreviewWhenDone
     if not original:
         # if there is no original camera
         # use the current camera as original
         original = rt.getNodeByName(camera)
     review_camera = rt.getNodeByName(camera)
     try:
         rt.viewport.setCamera(review_camera)
+        rt.preferences.playPreviewWhenDone = False
```

Comment:
I wouldn't expect a context manager that sets the camera to force a preference tweak like this. Note that `lib` should generalize functions ready for re-use. This seems to account for very specific use case instead of just setting the viewport camera. Might need a context manager of its own for preferences, for example:

```python

@contextlib.contextmanager
def preferences(settings):
    """Override rt.preferences key-values during context

    Example:
        >>> with preferences({"playPreviewWhenDone": False}):
        >>>     # in context here
        >>>     pass
    
    Arguments:
        settings (dict): key-value mapping of attributes to override.
    """
    original = {
        key: getattr(rt.preferences, key) for key in settings
    }
    try:
        for key, value in settings.items():
            setattr(rt.preferences, key, value)
        yield
    finally:
        for key, value in original.items():
            setattr(rt.preferences, key, value)
```
Example usage:

```python
with preferences({"playPreviewWhenDone": False}):
    # ....
```

Reply:
I would try in my side to see if it works.